frontend/src/app/main/main.component.ts

```diff
@@ -67,13 +73,25 @@ selector: any;
       })
       if (!dup) {
         this.questionService.saveQuestion(obj).subscribe((res) => {
-          // log error
+          obj["questionId"] = res.questionId;
+          this.counter++;
+          this.questions?.push(obj)
+          this.saveQuestions();
+          this.clearErrorMessage()
+          this.successMessage = "Successfully added your question as question " + res.questionId
+          formData.reset()
+        }, (err) => {
+          var errMessage = "An error occurred while adding your question!"
+          if (err.error) {
+            if (err.error.message.includes("duplicate key error")) {
```

Comment:
Should catch more errors?

Reply:
I think its fine for now honestly, as long as we mention in the report that only titles cannot be duplicated I suppose?